server.js

```diff
@@ -95,9 +105,34 @@ const createTarget = (req, options) => {
   }
 
   let XTargetURL = req.headers["x-target-url"];
-  // Support the eBay campaign.
-  if (XTargetURL && XTargetURL.startsWith("https://www.ebay.")) {
-    options.query.sub1 = "ebay";
+  let url, tld;
+  if (XTargetURL) {
+    try {
+      url = new URL(XTargetURL);
+      tld = psl.parse(url.hostname).tld;
+    } catch (ex) {
+      log.info("server", {msg: "Invalid URL passed for X-Target-URL: " + XTargetURL});
+    }
+
+    // TEMP WORKAROUND: if the region passed in the X-Region header doesn't
+    // match up with the region that the public suffix indicates, throw an error.
```

Comment:
nit: could be handy to reference the bug or issue blocking removing the workaround